vignettes/interventions.Rmd

```diff
@@ -0,0 +1,162 @@
+
+```{r simulate_chains_pop_control}
+sims <- simulate_summary(
+  nchains = 200, offspring_dist = "nbinom", stat_max = 99, mu = 0.9, size = 0.5
+)
+sims[is.infinite(sims)] <- 100
+ggplot(data.frame(x = sims), aes(x = x)) +
+  geom_histogram(breaks = seq(0, 100, by = 5), closed = "left") +
+  scale_x_continuous(breaks = c(0, 25, 50, 75, 100),
+                     labels = c(0, 25, 50, 75, ">99")) +
+  theme_bw()
+```
+
+## Individual-level control.
+
+In simulating population-level control we now apply the same reduction as before (25%) but instead of assuming that the mean is reduced we apply this such that 25% of individuals do not transmit further at all, whereas the remaining 75% generate offspring as in the uncontrolled case.
+
+To do this, we can no longer use the standard negative binomial distribution that comes with R.
+Instead, we define a random generator from a modified negative binomial distribution that includes our individual-level control as a `control` argument indicating the level of individual-level control (0: no control; 1: full control):
+
+```{r nbinom_ind_control}
+rnbinom_ind <- function(n, ..., control = 0) {
+  ## initialise number of offspring to 0
+  offspring <- rep(0L, n)
+  ## for each individual, decide whether they transmit further
+  transmits <- rbinom(n = n, prob = 1 - control, size = 1)
+  ## check if anyone transmits further
+  if (sum(transmits) > 0L) {
+    ## for those that transmit, sample from negative binomial with given
+    ## parameters
+    offspring[which(transmits == 1L)] <- rnbinom(n = n, ...)
+  }
+  return(offspring)
+}
+```
+
+Having defined this, we can generate simulations as before:
+
+```{r simulate_chains_ind_control}
+sims <- simulate_summary(
+  nchains = 200, offspring_dist = "nbinom_ind", stat_max = 99, mu = 1.2, 
+  size = 0.5, control = 0.25
+)
+sims[is.infinite(sims)] <- 100
+ggplot(data.frame(x = sims), aes(x = x)) +
+  geom_histogram(breaks = seq(0, 100, by = 5), closed = "left") +
+  scale_x_continuous(breaks = c(0, 25, 50, 75, 100),
+                     labels = c(0, 25, 50, 75, ">99")) +
+  theme_bw()
+```
+
+# Preventing superspreading events
+
+Another way of controlling a disease would be to prevent individuals from spreading to a large number of others, for example by preventing mass gatherings or, more generally, settings where superspreading events can occur.
+
+We can model this by truncating the offspring distribution at a certain size.
+This can be done, for example, using the [truncdist](https://cran.r-project.org/package=truncdist) R package.
+We use this to define a truncated negative binomial offspring distribution:
+
+```{r negbin_truncated}
+library("truncdist")
+rnbinom_truncated <- function(n, ..., max = Inf) {
+  return(rtrunc(n = n, spec = "nbinom", b = max, ...))
+}
+```
+
+We use this to simulate chains in a situation where the maximum of secondary cases that each infected person can generate is 10.
```

Comment:
```suggestion
We use this to simulate chains in a situation where the maximum of secondary cases that each infected person can generate is 10. In real life, this can be likened to a disease control strategy where gatherings are limited to 10 people.
```